### PR TITLE
[ci] add container audit

### PR DESCRIPTION
**Pull request by sausagee**
## Motivation
- Add nightly container audit and create GH issue when vuln is detected
- Add trivy to CI base image

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan
container audit canary 
https://github.com/sausagee/libra/runs/2505679622?check_suite_focus=true which filed issue https://github.com/sausagee/libra/issues/159

dev-setup canary locally 

ci base image canary https://github.com/diem/diem/runs/2505696820?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/daily.yml [hunk 1]
```diff
@@ -54,6 +54,48 @@ jobs:
...
+  audit-docker:
+    runs-on: ubuntu-latest
+    env:
+      IMAGES: "diem/validator diem/validator_tcb diem/init diem/tools"
+      TAG: devnet
+      AUDIT_SUMMARY_FILE: /tmp/summary
+    steps:
+      - uses: actions/checkout@v2
+      - name: install trivy as needed
+        run: |
```
Comment (rexhoffman) [inline .github/workflows/daily.yml:66]: dev-setup.sh will do this if you use the base image you've updated the dev-setup.sh to create.

Comment (sausagee) [reply .github/workflows/daily.yml:66]: yep, this was added to the dev-setup and and github base image below.  This was for canary before the base image built and published to Dockerhub.

File: scripts/dev_setup.sh [hunk 2]
```diff
@@ -209,6 +210,26 @@ function install_awscli {
...
 }
 
+function install_trivy {
+  VERSION=$(trivy --version | head -1)
+  if [[ "$VERSION" != "Version: ${TRIVY_VERSION}" ]]; then
+    if [[ $(uname -s) == "Darwin" ]]; then
+      install_pkg aquasecurity/trivy/trivy brew
+    else
+      install_pkg wget "$PACKAGE_MANAGER"
+      install_pkg apt-transport-https "$PACKAGE_MANAGER"
```
Comment (rexhoffman) [inline scripts/dev_setup.sh:220]: pretty sure this is debian/ubuntu specific and will break in alpine/elsewhere.   I recently cleaned up sccache to be ready for alpine.  🤷‍♂️

File: .github/workflows/daily.yml (hunk 1 above)
Comment (rexhoffman) [reply .github/workflows/daily.yml:66]: Cool, just checking.

File: scripts/dev_setup.sh (hunk 2 above)
Comment (sausagee) [reply scripts/dev_setup.sh:220]: let me double check the test results

**Comment by sausagee** [comment]
sccache is currently broken for the CI base image.  Fix is in https://github.com/diem/diem/pull/8324.

File: scripts/dev_setup.sh [hunk 3]
```diff
@@ -209,6 +210,19 @@ function install_awscli {
...
 }
 
+function install_trivy {
+  VERSION=$(trivy --version | head -1)
+  if [[ "$VERSION" != "Version: ${TRIVY_VERSION}" ]]; then
+    if [[ $(uname -s) == "Darwin" ]]; then
+      install_pkg aquasecurity/trivy/trivy brew
+    else
+      # Use trivy's script to downloads the binary based on OS and arch
+      curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b "${HOME}"/bin "v$TRIVY_VERSION"
```
Comment (rexhoffman) [inline scripts/dev_setup.sh:220]: dislike this method of installation.   Most of our github installs are unzips of releases.

**Comment by davidiw** [comment]
Old PR, feel free to re-open.

**Pull request closed without merging**